### ISSUE TITLE
question about model.py

### ISSUE THREAD

**Issue by SHITH-HUNG-TANG** [ctrasd/GA-ICDNet#1]
Hello, 

I have a question about the program you mentioned in your paper, I'm not sure if I understand it correctly, the following is my understanding.

According to the paper mentioned f_id + f_cov, but I see the program model.py line347~352, do other actions.
![image](https://user-images.githubusercontent.com/20163371/158011435-9d9e78a9-a511-4852-ac40-bd1cb57f6e8d.png)

```py
   features=[]
    for i in range(4):
        features.append(self.group_conv[i](x_bag_feature))
    features_bag=torch.cat([j for j in features], dim=1) # batch*32*1*1

    x=torch.cat([x_ident,features_bag],dim=1) #batch*128*1*1 
```

Thank you for your attention to this matter

**Comment by ctrasd** [comment]
I'm sorry that the equation on the figure is misleading. The true action is **concat**  rather than the **add** operation in math. Please follow the code.

Thank you for your attention to this work!

**Comment by SHITH-HUNG-TANG** [comment]
Thank you for your interpretation, it helps me a lot!

**Comment by SHITH-HUNG-TANG** [comment]
Hello, sorry for bothering you again.
Can I ask a question in model.py line349  x_bag_feature is 128-channel to split it , get [f_con1, f_con2, f_con3, f_con4]

but my understanding is use f_con (32-channel) to split it ,
![image](https://user-images.githubusercontent.com/20163371/159447887-693126f8-f172-4ea4-93b7-48c0e0e64022.png)

Thank you for your time and consideration

因為我英文不是非常好，想附上我中文原意

我看model.py line349  中使用的x_bag_feature 是 128-channel 但是我理解您的論文中使用的是f_con (32-channel) 去做切割出 [f_con1, f_con2, f_con3, f_con4]

我目前從程式的理解的流程圖長的像下面這個樣子，再麻煩指教謝謝您
![image](https://user-images.githubusercontent.com/20163371/159481111-dcb78dfe-1da1-43c1-9c7b-c1d2739eae3f.png)

我主要想問的是，為什麼不是使用x_bag[N,32,1,1]來作split而是使用 x_bag_feature[N,128,8,8]。不知道我的理解正不正確，造成您的困擾，萬分感謝。

另外，還有幾個小疑問想請教您，

1. 想問您在釋出的程式中有使用batch size 350，想請問設置batch size 350、300之間的差異有沒有什麼選擇上的建議，因為我實際運行了您的程式，發現設置350有時不會比300的效果差，但是最終論文的超參數設置為300。

2. 想請教您在這類型的模型訓練中，是不是都不會討論vaild Set ，只會分為train Set 和 test Set，然後選擇 test Set在rank1中得分最高的那次，作為最終的最佳權重，謝謝您。

真的的萬分感謝您，真的是一篇非常好的論文，讓我學到很多。

**Comment by ctrasd** [comment]
抱歉，最近因为疫情工作等等原因比较忙，回复的比较迟。。对，你理解的没有问题，代码是在特征图[N,128,8,8]时就切分了，这步当时主要是希望不同部分的特征图用独立的卷积层去过它们。我没有做在[N,32,1,1]和[N,128,8,8]两个地方切分的对比实验，你可以试试哪个效果更好。

1，batch size这个比较玄学。。我也没有过多的尝试。可能会有一些性能波动，但是这些都只是调参层面上的，只能说尝试尝试。。
2，目前re-id和步态任务基本都是没有val set的，基本都是取test set中最高的作为结果。

**Comment by SHITH-HUNG-TANG** [comment]
了解了解! 謝謝 @ctrasd ，謝謝您的解說以及您的程式，真的幫助我非常的多。

在這祝您身體健康、工作順利，特別感謝您抽空解答我的問題，萬分感謝。
